generated/cgo_helpers.go

```diff
@@ -242,11 +242,17 @@ func allocFilClearCacheResponseMemory(n int) unsafe.Pointer {
 
 const sizeOfFilClearCacheResponseValue = unsafe.Sizeof([1]C.fil_ClearCacheResponse{})
 
-// unpackPCharString represents the data from Go string as *C.char and avoids copying.
+// unpackPCharString copies the data from Go string as *C.char.
 func unpackPCharString(str string) (*C.char, *cgoAllocMap) {
+	allocs := new(cgoAllocMap)
+	defer runtime.SetFinalizer(allocs, func(a *cgoAllocMap) {
```

Comment:
Unrelated but something I came up with:
If you had function `GoFree` on the cgoAllocMap (and different alloc maps) that has the content of this lambda.
Then it can be shortened to:
`runtime.SetFinalizer(allocs, (*cgoAllocMap).GoFree)` while eliminating the need for a closure.

Reply:
See: https://play.golang.org/p/55AytJ-XjXz

Reply:
@Kubuxu that's a cool shortcut!